src/main/groovy/cz/alenkacz/gradle/marathon/deploy/DeployTask.groovy

```diff
@@ -0,0 +1,38 @@
+package cz.alenkacz.gradle.marathon.deploy
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+class DeployTask extends DefaultTask {
+    public DeployTask() {
+        group = 'publishing'
+        description = 'Deploys your application to Marathon'
+    }
+    @TaskAction
+    def deployToMarathon() {
+        project.exec { execSpec ->
+            if (!project.marathon.marathonUrl) {
```

Comment:
I would personally prefer to pass the instance of the extension to this class directly. So introduce property for the extension here and set it from `MarathonDeployPlugin`.